src/nirfsg/system_tests/test_system_nirfsg.py

```diff
@@ -0,0 +1,559 @@
+        script0 = '''script myScript0
+        wait until scriptTrigger0
+        generate mywaveform
+        end script'''
+        test_rfsg_device_session.write_script(script0)
+        test_rfsg_device_session.configure_software_start_trigger()
+        test_rfsg_device_session.configure_software_script_trigger('scriptTrigger0')
+        with test_rfsg_device_session.initiate():
+            test_rfsg_device_session.send_software_edge_trigger(nirfsg.SoftwareTriggerType.START, '')
+            test_rfsg_device_session.send_software_edge_trigger(nirfsg.SoftwareTriggerType.SCRIPT, 'scriptTrigger0')
+
+    @pytest.mark.skipif(sys.platform == "linux", reason="Function not supported on Linux OS")
+    def test_deembedding_table_with_s2p_file(self, test_rfsg_device_session):
+        test_rfsg_device_session.create_deembedding_sparameter_table_s2p_file('', 'myTable1', get_test_file_path('samples2pfile.s2p'), nirfsg.SparameterOrientation.PORT2)
+        test_rfsg_device_session.create_deembedding_sparameter_table_s2p_file('', 'myTable2', get_test_file_path('samples2pfile.s2p'), nirfsg.SparameterOrientation.PORT1)
+        test_rfsg_device_session.configure_deembedding_table_interpolation_linear('', 'myTable1', nirfsg.Format.MAGNITUDE_AND_PHASE)
+        test_rfsg_device_session.deembedding_port[''].deembedding_selected_table = 'myTable1'
+        with test_rfsg_device_session.initiate():
+            test_rfsg_device_session.check_generation_status()
+        test_rfsg_device_session.delete_deembedding_table('', 'myTable1')
+        test_rfsg_device_session.deembedding_port[''].deembedding_selected_table = 'myTable2'
+        with test_rfsg_device_session.initiate():
+            test_rfsg_device_session.check_generation_status()
+        test_rfsg_device_session.delete_all_deembedding_tables()
+        try:
+            test_rfsg_device_session.commit()
+            assert False
+        except nirfsg.Error as e:
+            assert e.code == -1074097772
+            assert 'The specified de-embedding table cannot be found' in e.description
+        test_rfsg_device_session.deembedding_port[''].deembedding_selected_table = ''
+        with test_rfsg_device_session.initiate():
+            test_rfsg_device_session.check_generation_status()
+
+    def test_read_and_download_waveform_from_file_tdms(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_generation_mode(nirfsg.GenerationMode.ARB_WAVEFORM)
+        test_rfsg_device_session.read_and_download_waveform_from_file_tdms('mywaveform', get_test_file_path('ValidWaveformTDMSFile.tdms'), 0)
+        waveformexists = test_rfsg_device_session.check_if_waveform_exists('mywaveform')
+        assert waveformexists is True
+
+    def test_wait_until_settled(self, test_rfsg_device_session):
+        test_rfsg_device_session.configure_rf(2e9, -5.0)
+        with test_rfsg_device_session.initiate():
+            test_rfsg_device_session.wait_until_settled(15000)
```

Comment:
Will this wait 15 seconds for a simulated device, too?
If so, that's way too long for this test.
Remember, we run the tests on all supported Pythons, from the earliest (3.9) to the latest (3.13). That's 5 versions. So we have to run this test 5 times.